tests/Feature/SharingWithAnonUserAndPublicSearchTest.php

```diff
@@ -0,0 +1,157 @@
+
+		$responseForRoot = $this->root_album_tests->get();
+		$responseForRoot->assertJson($this->generateExpectedRootJson(
+			null, $this->photoID1, null, $this->photoID1
+		));
+		$responseForRoot->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForRecent = $this->albums_tests->get(RecentAlbum::ID);
+		$responseForRecent->assertJson($this->generateExpectedSmartAlbumJson(
+			true,
+			$this->photoID1, [
+				$this->generateExpectedPhotoJson(static::SAMPLE_FILE_TRAIN_IMAGE, $this->photoID1, $this->albumID1),
+			]
+		));
+		$responseForRecent->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForStarred = $this->albums_tests->get(StarredAlbum::ID);
+		$responseForStarred->assertJson($this->generateExpectedSmartAlbumJson(
+			true,
+			$this->photoID1, [
+				$this->generateExpectedPhotoJson(static::SAMPLE_FILE_TRAIN_IMAGE, $this->photoID1, $this->albumID1),
+			]
+		));
+		$responseForStarred->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForTree = $this->root_album_tests->getTree();
+		$responseForTree->assertJson($this->generateExpectedTreeJson());
+		$responseForTree->assertJsonMissing(['id' => $this->albumID1]);
+		$responseForTree->assertJsonMissing(['id' => $this->photoID1]);
+		$responseForTree->assertJsonMissing(['id' => $this->photoID2]);
+
+		$this->albums_tests->get($this->albumID1, $this->getExpectedInaccessibleHttpStatusCode(), $this->getExpectedDefaultInaccessibleMessage(), self::EXPECTED_PASSWORD_REQUIRED_MSG);
+		$this->photos_tests->get($this->photoID1);
+		$this->photos_tests->get($this->photoID2, $this->getExpectedInaccessibleHttpStatusCode());
+	}
+
+	public function testPublicUnsortedPhotoAndPhotoInSharedAlbum(): void
+	{
+		$this->preparePublicUnsortedPhotoAndPhotoInSharedAlbum();
+
+		$responseForRoot = $this->root_album_tests->get();
+		$responseForRoot->assertJson($this->generateExpectedRootJson(
+			null, $this->photoID1, null, $this->photoID1
+		));
+		$responseForRoot->assertJsonMissing(['id' => $this->albumID1]);
+		$responseForRoot->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForRecent = $this->albums_tests->get(RecentAlbum::ID);
+		$responseForRecent->assertJson($this->generateExpectedSmartAlbumJson(
+			true,
+			$this->photoID1, [
+				$this->generateExpectedPhotoJson(static::SAMPLE_FILE_TRAIN_IMAGE, $this->photoID1, null),
+			],
+		));
+		$responseForRoot->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForStarred = $this->albums_tests->get(StarredAlbum::ID);
+		$responseForStarred->assertJson($this->generateExpectedSmartAlbumJson(
+			true,
+			$this->photoID1, [
+				$this->generateExpectedPhotoJson(static::SAMPLE_FILE_TRAIN_IMAGE, $this->photoID1, null),
+			],
+		));
+		$responseForStarred->assertJsonMissing(['id' => $this->photoID2]);
+
+		$responseForTree = $this->root_album_tests->getTree();
+		$responseForTree->assertJson($this->generateExpectedTreeJson());
+		$responseForTree->assertJsonMissing(['id' => $this->photoID2]);
```

Comment:
Shouldn't `albumID1` and `photoID1` be missing here as well?

Reply:
Good question. I need to double check. Photo 1 is public, starred and appears in the recent album. As the global option `public_search` is enabled, I assumed photo 1 would be shown as a thumbnail for the root (i.e. unsorted album) in the album tree. But maybe our album tree does never show a thumbnail for the root at all. I cannot remember by heart how it is implemented.

At least it would not be an error from the security perspective, _if_ photo 1 was shown as a thumbnail for the root.

But sure, if it isn't, then it should be added to the list in order to avoid confusion of reviewers.

Reply:
Speaking of thumbs, I noticed that if we explicitly specify a thumb for an album, it is returned to the front end even if the album is password-protected _and still locked_. This does not appear to be a regression (I could reproduce it with 4.4.0); still, we should fix it.

Reply:
> Speaking of thumbs, I noticed that if we explicitly specify a thumb for an album, it is returned to the front end even if the album is password-protected _and still locked_. This does not appear to be a regression (I could reproduce it with 4.4.0); still, we should fix it.

It is not a bug, it is a _feature_. :D
I am totally fine with that behavior personally and wanted to make a PR for it, so if it is already working, then even better.
The rationale being that the user is voluntarily setting the image. That being said, I would make this behavior more explicitly known to the user. :)

Reply:
I disagree. Yes, the user set an image as a preferred cover of an album, but they never agreed for that image being shown to every random Joe who happens to browse by.

I can set a cover image of a private album; does that mean that somehow that thumb should now be exposed to everybody, since I explicitly set it? Of course not. The access permissions take precedence.

I mean, sure, I can imagine a scenario where one might _want_ the current behavior (showing a "taster", if you will), but it must be possible to configure it; I also don't think that the current behavior should be the default.

Reply:
Sorry, I didn't want to resolve the conversation. I just hit the wrong button. But it is unresolved again.

I totally agree with @kamil4. This is a security bug and should be fixed. Actually I am disappointed (of myself) that I did not consider this case on my tests.